src/postprocess/DOS.jl

```diff
@@ -0,0 +1,41 @@
+# Compute densities of states
+
+# IDOS (integrated)
+# N(ε) = sum_n f_n = sum_n f((εn-ε)/T)
+# DOS
+# D(ε) = N'(ε)
+# LDOS (local)
+# LD = sum_n f_n ψn
+
+using ForwardDiff
+
+"""
+Total density of states at energy ε
+"""
+function DOS(ε, basis, orben; smearing=basis.model.smearing, T=basis.model.temperature)
+    D = zero(ε)
+    T != 0 || error("DOS only supports finite temperature")
+    for ik = 1:length(orben)
+        for iband = 1:length(orben[ik])
+            D += basis.kweights[ik]*ForwardDiff.derivative(ε -> smearing((orben[ik][iband]-ε)/T), ε)
+        end
+    end
+    D
+end
+
+"""
+Local density of states, in real space
+"""
+function LDOS(ε, basis, orben, Psi; smearing=basis.model.smearing, T=basis.model.temperature)
+    D = zeros(typeof(ε), basis.fft_size)
+    T != 0 || error("DOS only supports finite temperature")
+    for ik = 1:length(orben)
+        ψreal = G_to_r(basis, basis.kpoints[ik], Psi[ik])
+        for iband = 1:length(orben[ik])
+            D += basis.kweights[ik] *
+                ForwardDiff.derivative(ε -> smearing((orben[ik][iband]-ε)/T), ε) *
+                abs2.(ψreal[:, :, :, iband])
```

Comment:
I like for breaks such as this to have the operator as the first thing in the next line (especially for sums this is helpful), also spacing around operators :smile:.